apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/OpenApiDocs.java

```diff
@@ -0,0 +1,191 @@
+import java.util.Set;
+import org.hsqldb.jdbc.JDBCDataSource;
+
+/** Program to gather the OpenAPI yaml content for V1 and V2 and store. */
+public class OpenApiDocs {
+
+  /** The HSQL database url. */
+  private static final String INMEM_HSQL_DATABASE_URL = "jdbc:bfd-test:hsqldb:mem";
+
+  /** The version from the project pom. */
+  private final String projectVersion;
+
+  /** The destination directory for the output of the OpenAPI yaml file documents. */
+  private final String destinationDirectory;
+
+  /** The base server url. */
+  private String baseServerUrl;
+
+  /** The request specification to use with the api-docs requests. */
+  private RequestSpecification requestAuth;
+
+  /**
+   * Constructs an instance of OpenApiDocs for a given project version and destination directory.
+   *
+   * @param projectVersion @see #projectVersion
+   * @param destinationDirectory @see #destinationDirectory
+   */
+  private OpenApiDocs(String projectVersion, String destinationDirectory) {
+    this.projectVersion = projectVersion;
+    this.destinationDirectory = destinationDirectory;
+  }
+
+  /**
+   * Downloads the OpenAPI yaml from the api-docs V1 endpoint using a local E2E test server
+   * instance.
+   *
+   * @param apiVersion the bfd api version, either V1 or V2.
+   * @throws IOException when IO operations fail
+   */
+  void downloadApiDocs(String apiVersion) throws IOException {
+    var apiDocsUrl = String.format("%s/%s/fhir/api-docs", baseServerUrl, apiVersion.toLowerCase());
+    var response =
+        given()
+            .spec(requestAuth)
+            .when()
+            .get(apiDocsUrl)
+            .then()
+            .contentType("text/yaml")
+            .extract()
+            .asString();
+
+    writeFile(apiVersion, response);
+  }
+
+  /**
+   * Write the OpenAPI yaml content to a file.
+   *
+   * @param apiVersion either V1 or V2
+   * @param contents the OpenAPI yaml contents to write to the file
+   * @throws IOException when file operations fail
+   */
+  private void writeFile(String apiVersion, String contents) throws IOException {
+    var fileName =
+        String.format("%s/%s-OpenAPI-%s.yaml", destinationDirectory, apiVersion, projectVersion);
+    try (var fileWriter = new FileWriter(fileName)) {
+      fileWriter.write(contents);
+    }
+  }
+
+  /**
+   * Entry point, starts an E2E test server instance and downloads OpenAPI yaml.
+   *
+   * @param args The project version, working directory, and destination directory.
+   */
+  public static void main(String[] args) {
+
+    // Validate arguments.
+    validateArgs(args);
+
+    var projectVersion = args[0];
+    var workingDirectory = args[1];
+    var destinationDirectory = args[2];
+
+    // Update working directory so E2E test server instance can find properties.
+    System.setProperty("user.dir", workingDirectory);
+
+    // Set database url for in memory HSQL database
+    System.setProperty("its.db.url", INMEM_HSQL_DATABASE_URL);
```

Comment:
Initially I inherited from ServerRequiredTest and it wouldn't run as part of the build because Testcontainers couldn't find Docker to launch a container for the db.  I don't know if that happens with the in-mem HSQL,  I didn't realize that was an option until I started looking at the test utility classes.

Reply:
You need docker running when the test is running to use ServerRequiredTest, so maybe that was the issue (if you were seeing failures locally). I'm pretty sure we are able to run ServerRequiredTest tests from testcontainers normally in our CI process, so it should be doable.

We've also been trying to wean off using HSQL generally (to align with the app that uses postgres), so was also hoping to avoid that on the side so we could potentially remove that dependency one day.

Reply:
Resolving per discussion in standup.